src/armTemplates/compositeArmTemplate.ts

```diff
@@ -36,13 +42,13 @@ export class CompositeArmTemplate implements ArmResourceTemplateGenerator {
     return template;
   }
 
-  public getParameters(config: ServerlessAzureConfig) {
+  public getParameters(config: ServerlessAzureConfig, namer: (resource: ArmResourceType) => string) {
```

Comment:
It doesn't feel right that the `getParameters(...)` function now requires the function to name the resource.  Since you have already added `ArmResourceType` enum - what if we have the ARM resource classes just take a dependency on the naming service?  And call the naming service as needed to generate the name?